### PR TITLE
Fix number of choices when splitting a troop into multiple troops

### LINKED ISSUE #3587: Hero screen, sometimes the number of choices for splitting a troop into multiple troops does not match the possibilities (0.9.3 3574)
In fheroses2, sometimes the number of choices for splitting a troop into multiple troops does not match the possibilities.
In the example, the number of splitting "4" is shown, while there are only two free slots in the hero's army.

In fheroses2:

Split bug 01

Split bug 02

https://user-images.githubusercontent.com/43583994/120362347-41be9e00-c30b-11eb-8637-7f5b8232638f.mp4

Here is the save file:
[Before split bug 01.zip](https://github.com/ihhub/fheroes2/files/6577865/Before.split.bug.01.zip)

### PR DESCRIPTION
**Pull request by oleg-derevenetz**
fix #3587

### CODE REVIEW

File: src/fheroes2/army/army_bar.cpp [hunk 1]
```diff
@@ -359,7 +362,7 @@ bool ArmyBar::ActionBarLeftMouseSingleClick( ArmyTroop & troop )
         ArmyTroop * selectedTroop = GetSelectedItem();
         assert( selectedTroop != nullptr );
 
-        const bool isSameTroopType = troop.GetID() == selectedTroop->GetID();
+        const bool isSameTroopType = troop.isValid() && troop.GetID() == selectedTroop->GetID();
```
Comment (ihhub) [inline src/fheroes2/army/army_bar.cpp:365]: Hi @oleg-derevenetz , I assume that this and change on line 450 are the only changes needed to fix the original issue. Is it correct?

Comment (oleg-derevenetz) [reply src/fheroes2/army/army_bar.cpp:365]: Hi @ihhub Lines 40 and 76 also. Moving local functions to anonymous namespace is totally optional.

Comment (oleg-derevenetz) [reply src/fheroes2/army/army_bar.cpp:365]: @ihhub There is a separate commit for this, I can roll it back if you wish.

Comment (ihhub) [reply src/fheroes2/army/army_bar.cpp:365]: No, it's fine.

**Comment by ihhub** [comment]
@oleg-derevenetz , a bunch of thanks for fixing this issue!

**Pull request merged**